tap/tests/test_parser.py

```diff
@@ -196,18 +197,83 @@ def test_parses_file(self):
         self.assertEqual('plan', lines[0].category)
         self.assertEqual('test', lines[1].category)
         self.assertTrue(lines[1].ok)
+        self.assertIsNone(lines[1].yaml_block)
         self.assertEqual('test', lines[2].category)
         self.assertFalse(lines[2].ok)
 
-    @mock.patch('tap.parser.sys')
-    def test_parses_stdin(self, mock_sys):
-        mock_sys.stdin.__iter__.return_value = iter([
-            '1..2\n',
-            'ok 1 A passing test\n',
-            'not ok 2 A failing test\n',
-        ])
-        mock_sys.stdin.__enter__.return_value = None
-        mock_sys.stdin.__exit__.return_value = None
+    def test_parses_yaml(self):
```

Comment:
These tests for optional functionality are failing. How can I test both regression (i.e. behaviors with modules not installed) and new functionality?